src/test/regress/citus_tests/common.py

```diff
@@ -24,9 +25,19 @@ def initialize_temp_dir_if_not_exists(temp_dir):
     os.chmod(temp_dir, 0o777)
 
 
+def parallel_run(function, items, *args, **kwargs):
```

Comment:
As far as I remember, we already run the arbitrary configs parallel locally. We preferred single parallelism locally mostly because our laptops couldn't handle more than 2 :
```
# will run 4 configs in parallel
make check-arbitrary-configs parallel=4

```


And, this commit increased the parallelism on the CI with another approach: https://github.com/citusdata/citus/commit/56abd3d5019f20ee5876ad785973ea7584ac0e08


Which we can further increase. 

So, what is the main goal of this change?  Can the above be used?

Reply:
Currently we can only run different configs in parallel. However, when working on a feature or trying to fix a bug this is not important. In those cases you simply want to run a single test file on a single config. And you want to run that every time you made a change to the code that you think fixes the issue.

This PR allows parallelising running of bash commands. So `initdb` and `pg_ctl start` is run in parallel for all nodes in the cluster. Instead of one waiting for the other.

When you run this command currently pretty much nothing runs in parallel:
```
make -C src/test/regress/ check-arbitrary-base CONFIGS=CitusDefaultClusterConfig EXTRA_TESTS=prepared_statements_1
```